systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: How many states to retrieve. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 1000
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateDescriptionListResponse'
+        401:
+          $ref: '#/responses/Unauthorized'            
+        default:
+          $ref: '#/responses/Error'
+    post:
+      operationId: CreateState
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      tags: [states]
+      summary: Create a new system state
+      description: Create a new System State by providing metadata and its content. rawState property will be used if it has a non-null value, feeds and packages will be used otherwise.
+      parameters:
+        - in: body
+          name: state
+          description: The state to create.
+          required: true
+          schema:
+            $ref: '#/definitions/StateRequest'
+      responses:
+        201:
+          description: Created
+          schema:
+            $ref: '#/definitions/StateResponse'
+        400:
+          $ref: '#/responses/BadRequest'            
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          $ref: '#/responses/Conflict'
+        default:
+          $ref: '#/responses/Error'
+  /v1/states/{id}:
+    get:
+      operationId: GetState
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [states]
+      summary:  Get system state
+      description: Returns the metadata, the feeds and packages from the content of the state. If the state contains only feeds and packages entries the containsExtraOperations from response will be false, otherwise it will be true. The export route could be used in order to get the whole content of the state.
+      parameters:
+        - name: id
+          in: path
+          description: ID of the state to return.
+          required: true
+          type: string
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateResponse'
+        401:
+          $ref: '#/responses/Unauthorized'          
+        404:
+          $ref: '#/responses/NotFound'
+        default:
+            $ref: '#/responses/Error'
+    patch:
+      operationId: UpdateState
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      consumes:
+        - application/merge-patch+json      
+      tags: [states]
+      summary: Update a state
+      description: Update a state by providing specific changes of the state. The method overrides any state properties with what is contained in the request. In order to ignore properties update, leave the value off the request. If rawState is specified it will have priority against packages/feeds.
```

Comment:
"Update a state by providing specific changes to the state. The method overrides all state properties with those contained in the request. Therefore, if you want to ignore a properties update, leave the value off the request. If rawState is specified in the request, it will have priority over packages/feeds."?